### PR TITLE
bug-1812771: clean up processor rules for java/android things

### PR DESCRIPTION
**Pull request by willkg**
This moves some java/android related processor rule code into separate modules. It makes it a little easier to find rules.

### CODE REVIEW

File: socorro/mozilla_rulesets.py [hunk 1]
```diff
@@ -91,14 +92,15 @@
     OutOfMemoryBinaryRule(),
     PHCRule(),
     BreadcrumbsRule(schema=get_schema("processed_crash.schema.yaml")),
-    JavaProcessRule(),
```
Comment (willkg) [inline socorro/mozilla_rulesets.py:94]: "JavaProcessRule" isn't wildly helpful for what it does. It's a rule focusing on the `JavaStackTrace` annotation and processing that specifically. So I renamed it.

File: socorro/processor/rules/android.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    def predicate(self, raw_crash, dumps, processed_crash, tmpdir, status):
+        cpu_arch = processed_crash.get("cpu_arch", "unknown")
+        return cpu_arch == "unknown" and "Android_CPU_ABI" in raw_crash
+
+    def action(self, raw_crash, dumps, processed_crash, tmpdir, status):
+        android_cpu_abi = raw_crash["Android_CPU_ABI"]
+        cpu_arch = self.ANDROID_CPU_ABI_MAP.get(android_cpu_abi, android_cpu_abi)
+
+        processed_crash["cpu_arch"] = cpu_arch
```
Comment (willkg) [inline socorro/processor/rules/android.py:35]: `CPUInfoRule` sets `cpu_info` to "unknown" and `cpu_count` to "0" both of which aren't helpful, so we don't set them here.

File: socorro/processor/rules/general.py [hunk 3]
```diff
@@ -96,13 +96,6 @@ class CPUInfoRule(Rule):
 
     """
 
-    # Map of Android_CPU_ABI values to cpu_arch values
-    ANDROID_CPU_ABI_MAP = {
-        "armeabi-v7a": "arm",
-        "arm64-v8a": "arm64",
-        "x86_64": "amd64",
-    }
-
```
Comment (willkg) [inline socorro/processor/rules/general.py:105]: Oops--this needs a predicate so it only kicks in if `json_dump.system_info.cpu_info` exists.

Comment (willkg) [reply socorro/processor/rules/general.py:105]: Actually, scratch that--the way I have it now, it does exactly what the old code did.

File: socorro/processor/rules/android.py (hunk 2 above)
Comment (willkg) [reply socorro/processor/rules/android.py:35]: Actually, scratch that--the way I have it now, it does exactly what the old code did.

File: socorro/schemas/processed_crash.schema.yaml [hunk 4]
```diff
@@ -2848,12 +2848,16 @@ properties:
...
+      For Android, this is `Android` plus the sdk version.
     type: string
     permissions: ["public"]
   os_version:
-    description: >
+    description: |
       Version of the operating system. This comes from `system_info.os_ver`
       from the stackwalker output of the minidump.
+
+      For Android, this is the sdk version.
```
Comment (willkg) [inline socorro/schemas/processed_crash.schema.yaml:2860]: Oops--this shouldn't change in this PR.

**Comment by willkg** [comment]
There's some other reorganization work I want to do at some point, but I limited this cleanup work to the minimum to make the next change easier.

**Review by relud — APPROVED** [review]

**Comment by willkg** [comment]
Thank you!

**Pull request merged**